routes/post.js

```diff
@@ -0,0 +1,123 @@
+const http = require("http");
+const express = require("express")
+const cookieParser = require("cookie-parser")
+const bodyParser = require("body-parser")
+const ObjectId = require('mongoose').Types.ObjectId;
+const { Schema,
+    userSchema,
+    User,
+    connect,
+    sessionSchema,
+    Session,
+    postSchema,
+    Post,
+    replySchema,
+    Reply } = require("../database.js");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+    try {
+        if (!req.session) {
+            res.clearCookie('session');
+            res.redirect('/');
+            return;
+        }
+        const currentUser = await User.findById(req.session.user);
```

Comment:
Notice how the very first thing you do after every session check is getting the user? That would have been a clue that getting the user should be moved to a middleware, since you already have a related middleware that already fetches the user anyway, you should have moved this logic there.